validation/src/main/java/dk/kvalitetsit/itukt/validation/repository/cache/StamdataCacheImpl.java

```diff
@@ -2,36 +2,51 @@
 
 import dk.kvalitetsit.itukt.common.configuration.CacheConfiguration;
 import dk.kvalitetsit.itukt.common.repository.cache.CacheLoader;
-import dk.kvalitetsit.itukt.validation.repository.StamDataRepositoryAdaptor;
+import dk.kvalitetsit.itukt.validation.repository.Repository;
 import dk.kvalitetsit.itukt.validation.service.model.StamData;
 
-import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-public class StamdataCacheImpl implements StamdataCache, CacheLoader {
+public class StamdataCacheImpl implements Cache<Long, StamData>, CacheLoader {
 
     private final CacheConfiguration configuration;
-    private final StamDataRepositoryAdaptor concreteStamDataRepository;
-    private Map<Long, StamData> drugIdToClauseMap = new HashMap<>();
+    private final Repository<StamData> repository;
+    private volatile Map<Long, StamData> entries = Map.of();
 
-    public StamdataCacheImpl(CacheConfiguration configuration, StamDataRepositoryAdaptor concreteStamDataRepository) {
+    public StamdataCacheImpl(CacheConfiguration configuration, Repository<StamData> repository) {
         this.configuration = configuration;
-        this.concreteStamDataRepository = concreteStamDataRepository;
+        this.repository = repository;
+    }
+
+    private static StamData resolveConflict(StamData x, StamData y) {
+        var clauses = Stream.concat(x.clauses().stream(), y.clauses().stream()).collect(Collectors.toSet());
+        return new StamData(x.drug(), clauses);
+    }
+
+    private <T> Map<T, StamData> toMap(List<StamData> entry, Function<StamData, T> identity) {
+        return entry.stream().collect(Collectors.toMap(
+                identity,
+                Function.identity(),
+                StamdataCacheImpl::resolveConflict
+        ));
```

Comment:
identity parameteren er overflødig

```suggestion
    private Map<Long, StamData> toMap(List<StamData> entries) {
        return entries.stream().collect(Collectors.toMap(
                entry -> entry.drug().id(),
                Function.identity(),
                StamdataCacheImpl::resolveConflict
        ));
    }
```